### PR TITLE
[CBRD-20448] New system parameter use_compression

### PR DESCRIPTION
**Pull request by funkydvd**
I added a new system parameter, prm_use_compression, which works in the following way: if it is 0, the compression is disabled (there is no real compression, the compression length is set to 0). If it is under 32, it will be set to 32, otherwise it will be a value between 32 and 255, which will be the minimum string length for compression.
I also modified the parameter log_compress to be not only for enabling or disabling the log compression, but also for setting a threshold for the minimum size to compress a log.

### CODE REVIEW

**Comment by lucianlazar2211** [comment]
It looks great to me.

**Comment by eseokoh** [comment]
I agree with the change to only allow enable/disable mode. 
One change I'd like to ask you is to rename the parameter, "use_compression". Since there's no objective of "compression", it is unclear. 
Candidates are "enable_string_compression", "enable_long_string_compression", "string_compression", "long_string_compressIon". 
Please propose a nice name :)

**Comment by eseokoh** [comment]
I have a concern on performance.
Too many function calls to retrieve the parameter may hurt performance, since the functions are primitive functions and they are frequently called. 
What about introducing a global for the param? I don't think the parameter will be tuned during runtime, especially on production systems.

**Comment by funkydvd** [comment]
I reverted the changes and made a global for storing the param.

File: src/transaction/boot_sr.c [hunk 1]
```diff
@@ -3231,6 +3231,8 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
   common_ha_mode = prm_get_integer_value (PRM_ID_HA_MODE);
 #endif /* SERVER_MODE */
 
+  pr_Enable_string_compression = prm_get_bool_value (PRM_ID_ENABLE_STRING_COMPRESSION);
```
Comment (eseokoh) [inline src/transaction/boot_sr.c]: Please move it to the block from line 3333:

```C
#if defined(SERVER_MODE)
  if (sysprm_load_and_init (boot_Db_full_name, NULL) != NO_ERROR)
    {
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_BO_CANT_LOAD_SYSPRM, 0);
      error_code = ER_BO_CANT_LOAD_SYSPRM;
      goto error;
    }

  ...
```

**Comment by eseokoh** [comment]
sorry again. 

Please move the line after `sysprm_load_and_init`. 

```C
   if (sysprm_load_and_init ...
     ...

   pr_Enable_string_compression = prm_get_bool_value (PRM_ID_ENABLE_STRING_COMPRESSION);

```

**Comment by eseokoh** [comment]
Thank you :)

**Comment by eseokoh** [comment]
You can merge the fix if you don't have a loose-end. 
Thank you :)

**Comment by funkydvd** [comment]
I added a new system parameter named PRM_ID_ENABLE_STRING_COMPRESSION, which is used for turning the compression on and off. The string format remains the same (under 255 it has the length on a byte then the string, after 255 it has a byte containing 255, the compressed length, the decompressed length, the string and the alignment), but the compressed length is set to 0, so the string is stored in its original size.

**Pull request merged**